src/main/java/com/amazon/aws/iot/greengrass/semver/Range.java

```diff
@@ -0,0 +1,486 @@
+                } else {
+                    // nothing is forbidden
+                    ret = "*";
+                }
+            } else if (!gtlt.isEmpty() && anyX) {
+                // we know patch is an x, because we have any x at all.
+                // replace X with 0
+                if (xm) {
+                    m = "0";
+                }
+                p = "0";
+
+                if (">".equals(gtlt)) {
+                    // >1 => >=2.0.0
+                    // >1.2 => >=1.3.0
+                    gtlt = ">=";
+                    if (xm) {
+                        M = String.valueOf(Integer.parseInt(M) + 1);
+                        m = "0";
+                        p = "0";
+                    } else {
+                        m = String.valueOf(Integer.parseInt(m) + 1);
+                        p = "0";
+                    }
+                } else if ("<=".equals(gtlt)) {
+                    // <=0.7.x is actually <0.8.0, since any 0.7.x should
+                    // pass.  Similarly, <=7.x is actually <8.0.0, etc.
+                    gtlt = "<";
+                    if (xm) {
+                        M = String.valueOf(Integer.parseInt(M) + 1);
+                    } else {
+                        m = String.valueOf(Integer.parseInt(m) + 1);
+                    }
+                }
+
+                if ("<".equals(gtlt)) {
+                    pr = "-0";
+                }
+
+                ret = String.format("%s%s.%s.%s%s", gtlt, M, m, p, pr);
+            } else if (xm) {
+                ret = String.format(">=%s.0.0%s <%s.0.0-0", M, pr, Integer.parseInt(M) + 1);
+            } else if (xp) {
+                ret = String.format(">=%s.%s.0%s <%s.%s.0-0", M, m, pr, M, Integer.parseInt(m) + 1);
+            }
+
+            return ret;
+        }
+
+        // ~, ~> --> * (any, kinda silly)
+        // ~2, ~2.x, ~2.x.x, ~>2, ~>2.x ~>2.x.x --> >=2.0.0 <3.0.0-0
+        // ~2.0, ~2.0.x, ~>2.0, ~>2.0.x --> >=2.0.0 <2.1.0-0
+        // ~1.2, ~1.2.x, ~>1.2, ~>1.2.x --> >=1.2.0 <1.3.0-0
+        // ~1.2.3, ~>1.2.3 --> >=1.2.3 <1.3.0-0
+        // ~1.2.0, ~>1.2.0 --> >=1.2.0 <1.3.0-0
+        static String replaceTildes(String comp) {
+            return Arrays.stream(S_PLUS.split(comp.trim(), -1)).map(FreeFunctions::replaceTilde)
+                    .collect(Collectors.joining(" "));
+        }
+
+        private static String replaceTilde(String s) {
+            Pattern r = SemVerRegexes.re.get(SemVerRegexes.t.get("TILDE"));
+            Matcher mm = r.matcher(s);
+            if (!mm.find()) {
+                return s;
+            }
+
+            String M = mm.group(1);
+            String m = mm.group(2);
+            String p = mm.group(3);
+            String pr = mm.group(4);
+
+            String ret;
+            if (isX(M)) {
+                ret = "";
+            } else if (isX(m)) {
+                ret = String.format(">=%s.0.0 <%s.0.0-0", M, Integer.parseInt(M) + 1);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;+&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 327. Consider formatting this data with &#39;%d&#39; instead.